src/Services/src/Threading/ThreadingService.cs

```diff
@@ -5,13 +5,47 @@ namespace Tanzu.Toolkit.Services.Threading
 {
     public class ThreadingService : IThreadingService
     {
-        public ThreadingService()
+        private bool _isPolling = false;
+        private IUiDispatcherService _dispatcherService;
+
+        public bool IsPolling
+        {
+            get => _isPolling;
+
+            set { _isPolling = value; }
+        }
+
+        public ThreadingService(IUiDispatcherService dispatcherService)
         {
+            _dispatcherService = dispatcherService;
         }
 
         public void StartTask(Func<Task> method)
         {
             Task.Run(method);
         }
+
+        public void StartUiBackgroundPoller(Action<object> pollingMethod, object methodParam, int intervalInSeconds)
+        {
+            if (!IsPolling)
```

Comment:
can 2 threads get here at the same same, both think IsPolling is false, and both then set to true?

Reply:
In theory, yeah... at the moment though this method is only ever called once (i.e. only by 1 thread, after the very first login is done, and then the poller just keeps running for the rest of the app's lifetime). 
I'm happy to make this method more thread-safe but I can't imagine a scenario where it would need to be -- did you have one in mind?

Reply:
Sounds like it'k OK as is.